aip/general/0236.md

```diff
@@ -0,0 +1,349 @@
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "StopPreviewPolicyExperimentMetadata"
+  };
+}
+
+message StopPreviewPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+}
+
+message StopPreviewPolicyExperimentMetadata {}
+```
+- Admin flow API to stop previewing a resource **should** be implemented.
+  - Remains declarative friendly as only an output only field is modified.
+- If stopPreview is called specifically on a resource which is not actively being previewed, then it **should** return an error message.
+- When the API is called, PreviewMetadata.state **must** be set to INACTIVE and PreviewMetadata.stop_time **should** be set to the current time.
+- stopPreview is not idempotent, calling stop twice on the same resource **should** return an error.
+- If the API has an error, no fields in PreviewMetadata **should** change.
+- Etags **must** be used to guarantee resource freshness.
+
+#### commit
+
+A new API called commit **may** be implemented to promote the experimental configuration of a resource to the effective version and perform necessary cleanup. It will copy the fields-under-test from the experiment into the parent and then delete the experiment.
+
+This is a convenience API which is not declarative friendly. The alternative supported for declarative clients is to manually copy fields from an experiment onto the live policy then delete the experiment.
+
+```proto
+rpc CommitPolicyExperiment(CommitPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:commit"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "google.protobuf.Empty"
+    metadata_type: "CommitPolicyExperimentMetadata"
+  };
+}
+
+message CommitPolicyExperimentRequest {
+  string name = 1;
+  string etag = 2;
+  string parent_etag = 3; // optional.
+}
+
+message CommitPolicyExperimentMetadata {}
+```
+- Commit **must** atomically copy the fields-under-test from the experiment into the live policy then deletes the experiment.
+- If any experiment fails to be committed, the preview **should** be conitnued and the policy **should** not be updated.
+- Commit is not idempotent and calling commit twice on the same experiment **should** return an error.
+- Commit **may** be called at any point of time for an experiment resource being used for preview mode, either when the logs are actively being generated or after they are stopped. It has the same behavior in each case.
+- Etags **must** be used to guarantee resource freshness.
+
+### TTL of the Preview
+
+There is no recommendation to pass the TTL of a preview to the data plane which. Instead, the TTL can be controlled by the user of the APIs (direct API caller, Terraform, etc) by using the startPreview and stopPreview methods with a delay induced in between.
+
+### Logging
```

Comment:
I kind of feel like logging has to be clarified in an AIP around policy resources overall, so this section is consistent with that.

But do you think this guidance is critical?